File: Backend/Models/screeningForm.js
import mongoose from 'mongoose';

const screeningFormSchema = new mongoose.Schema({
    screeningId:{
        type: String,
        unique: true,
        required: true
    },
    interviewDate:{
        type: Date,
        required: true,
        default: Date.now
    },
    healthFacility:{
        type: String,
        enum: ['Bondo', 'Siaya', 'Kuoyo', 'Lumumba'],
        required: true
    },
    DoB:{
        type: Date,
        min: '1970-01-01',
        max: '2015-12-31',
        required: true
    },
    Age:{
        months:{
            type: Number,
            required: true
        },
        years:{
            type: Number,
            required: true
        }
    },
    height:{
        type: Number,
        required: true
    },
    weight:{
        type: Number,
        required: true
    },
    vitalSigns:{
        temperature:{
            value:{
                type: Number,
                required: true
            },
            location:{
                type: String,
                enum: ['Axillary', 'Oral', 'Tympanic'],
                required: true
            }
        },
        respiratoryRate:{
            type: Number,
            required: true
        },
        pulseRate:{
            type: Number,
            required: true
        },
        bloodPressure:{
            systolic:{
                type: Number,
                required: true
            },
            diastolic:{
                type: Number,
                required: true
            }
        }
    },
    lastMenstrualPeriod:{
        date:{
            type: Date,
            default: Date.now
        },
        unknown:{
            type: Boolean,
            default: false
        }
    },
    fundalHeight:{
        type: Number,
        required: true
    },
    inclusionCriteria:{
        residentWithin15km:{
            type: String,
            enum: ['Yes', 'No'],
            required: true
        },
        pregnancyConfirmed:{
            type: String,
            enum: ['Yes', 'No'],
            required: true
        },
        gestationLessThan31Weeks:{
            type: String,
            enum: ['Yes', 'No'],
            required: true
        },
        consentsToHIVTesting:{
            type: String,
            enum: ['Yes', 'No'],
            required: true
        },
        willingToDeliverAtStudyHospital:{
            type: String,
            enum: ['Yes', 'No'],
            required: true
        }
    },
    exclusionCriteria:{
        multiplePregancy:{
            type: String,
            enum: ['Yes', 'No', "Don't Know"],
            required: true
        },
        fisturaRepairOrSpinalDeformity:{
            type: String,
            enum: ['Yes', 'No', "Don't Know"],
            required: true
        },
        unableToGiveInformedConsent:{
            type: String,
            enum: ['Yes', 'No'],
            required: true
        }
    },
    eligibility:{
        meetsAllCriteria:{
            type: String,
            enum: ['Yes', 'No'],
            required: true
        },
        consentedToParticipate:{
            type: String,
            enum: ['Yes', 'No'],
            required: function (){
                return this.eligibility && this.eligibility.meetsAllCriteria === "Yes"
            }
        },
        reasonForRefusal:{
            type: String,
            enum: ['Needs to consult', 'Other', null],
            default: null,
            required: function (){
                return this.eligibility && this.eligibility.consentedToParticipate === "No"
            }
        }
    },
    createdAt:{
        type: Date,
        default: Date.now
    },
    updatedAt:{
        type: Date,
        default: Date.now
    }
});

const screeningForm = mongoose.model('screeningForm', screeningFormSchema);

export default screeningForm;

File: api/Models/ScreeningForm.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace KemriApi.Models
{    public class ScreeningForm
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("UserInitials")]
        public string UserInitials { get; set; } = string.Empty;

        [BsonElement("screeningId")]
        public string ScreeningId { get; set; } = string.Empty;

        [BsonElement("interviewDate")]
        public DateTime InterviewDate { get; set; } = DateTime.UtcNow;

        [BsonElement("healthFacility")]
        public string HealthFacility { get; set; } = string.Empty;

        [BsonElement("DoB")]
        public DateTime DoB { get; set; }

        [BsonElement("Age")]
        public AgeInfo Age { get; set; } = new();

        [BsonElement("height")]
        public double Height { get; set; }

        [BsonElement("weight")]
        public double Weight { get; set; }

        [BsonElement("BMI")]
        public double BMI { get; set; }

        [BsonElement("vitalSigns")]
        public VitalSignsInfo VitalSigns { get; set; } = new();

        [BsonElement("lastMenstrualPeriod")]
        public LmpInfo LastMenstrualPeriod { get; set; } = new();

        [BsonElement("fundalHeight")]
        public double FundalHeight { get; set; }

        [BsonElement("inclusionCriteria")]
        public InclusionCriteriaInfo InclusionCriteria { get; set; } = new();

        [BsonElement("exclusionCriteria")]
        public ExclusionCriteriaInfo ExclusionCriteria { get; set; } = new();

        [BsonElement("eligibility")]
        public EligibilityInfo Eligibility { get; set; } = new();

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [BsonElement("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }

    public class LmpInfo
    {
        [BsonElement("date")]
        public DateTime Date { get; set; } = DateTime.UtcNow;

        [BsonElement("unknown")]
        public bool Unknown { get; set; }
    }

    public class InclusionCriteriaInfo
    {
        [BsonElement("residentWithin15km")]
        public string ResidentWithin15km { get; set; } = "No";

        [BsonElement("pregnancyConfirmed")]
        public string PregnancyConfirmed { get; set; } = "No";

        [BsonElement("gestationLessThan31Weeks")]
        public string GestationLessThan31Weeks { get; set; } = "No";

        [BsonElement("consentsToHIVTesting")]
        public string ConsentsToHIVTesting { get; set; } = "No";

        [BsonElement("willingToDeliverAtStudyHospital")]
        public string WillingToDeliverAtStudyHospital { get; set; } = "No";
    }

    public class ExclusionCriteriaInfo
    {
        [BsonElement("multiplePregancy")]
        public string MultiplePregancy { get; set; } = "No";

        [BsonElement("fisturaRepairOrSpinalDeformity")]
        public string FisturaRepairOrSpinalDeformity { get; set; } = "No";

        [BsonElement("unableToGiveInformedConsent")]
        public string UnableToGiveInformedConsent { get; set; } = "No";
    }

    public class EligibilityInfo
    {
        [BsonElement("meetsAllCriteria")]
        public string MeetsAllCriteria { get; set; } = "No";

        [BsonElement("consentedToParticipate")]
        public string ConsentedToParticipate { get; set; } = "No";

        [BsonElement("reasonForRefusal")]
        public string? ReasonForRefusal { get; set; }
    }
}
